crates/ln-dlc-node/src/node/mod.rs

```diff
@@ -770,6 +771,29 @@ where
             settings,
         })
     }
+
+    pub fn update_ldk_settings(&self, ldk_config: UserConfig) {
```

Comment:
🤔 shouldn't we differentiate between public and private channels here? Otherwise we will always have on or the other if that gets executed.

Reply:
Good question! What this PR adds shouldn't have an effect on whether channels are public or private because that is controlled by `ChannelHandshakeConfig::announced_channel` which is not controlled by `ChannelConfig`. In general, I'm not sure there is a mechanism to turn a private channel into a public one.

Reply:
In relation to updating the `UserConfig` in general, we already have a default value for `announced_channel` for both coordinator and app. In the case of the coordinator we default to public channels and only override this when opening JIT channels in the event handler.

In theory we could change this default value with this API, but that would be weird. It's actually currently impossible to change it via the coordinator settings, which only support updating the `forwarding_fee_proportional_millionths` value:

https://github.com/get10101/10101/blob/2e574bf83ed40c48b6e6b72b7bdb1155c2c606c7/coordinator/src/settings.rs#L99-L110